json.ts

```diff
@@ -0,0 +1,1856 @@
+ */
+export type CloneableErrorTypes = keyof CloneableErrors;
+
+/**
+ * A representation of {@linkcode Error}s that can be stored as Deno KV values.
+ * The value is set to a serialized version of the value. Instances that are
+ * not one of the specified types, but inherit from `Error` will be serialized
+ * as `Error`.
+ */
+export interface KvErrorJSON<
+  ErrorType extends CloneableErrorTypes = CloneableErrorTypes,
+> {
+  /**
+   * The type of the error, which is one of the cloneable error types.
+   */
+  type: ErrorType;
+  /**
+   * The value of the error, which is a JSON serialized version of the error.
+   */
+  value: {
+    message: string;
+    cause?: KvValueJSON | undefined;
+    stack?: string | undefined;
+  };
+}
+
+/**
+ * A representation of a {@linkcode Deno.KvU64} value. The value is the string
+ * representation of the unsigned integer.
+ */
+export interface KvKvU64JSON {
+  /**
+   * The type of the value, which is always `"KvU64"`.
+   */
+  type: "KvU64";
+  /**
+   * The string representation of the unsigned integer value.
+   */
+  value: string;
+}
+
+/**
+ * A representation of a {@linkcode Map} Deno KV value. The value is an array
+ * of map entries where is map entry is a tuple of a JSON serialized key and
+ * value.
+ */
+export interface KvMapJSON {
+  /**
+   * The type of the value, which is always `"Map"`.
+   */
+  type: "Map";
+  /**
+   * The JSON serialized version of the map entries.
+   */
+  value: readonly [key: KvValueJSON, value: KvValueJSON][];
+}
+
+/**
+ * A representation of a {@linkcode null} Deno KV value. The value is `null`.
+ */
+export interface KvNullJSON {
+  /**
+   * The type of the value, which is always `"null"`.
+   */
+  type: "null";
+  /**
+   * The value of the value, which is always `null`.
+   */
+  value: null;
+}
+
+/**
+ * A representation of a {@linkcode object} Deno KV value. The value is a JSON
+ * serialized version of the value.
+ */
+export interface KvObjectJSON {
+  /**
+   * The type of the value, which is always `"object"`.
+   */
+  type: "object";
+  /**
+   * The JSON serialized version of the object.
+   */
+  value: { [key: string]: KvValueJSON };
+}
+
+/**
+ * A representation of a {@linkcode RegExp} Deno KV value. The value is a string
+ * representation of the regular expression value.
+ */
+export interface KvRegExpJSON {
+  /**
+   * The type of the value, which is always `"RegExp"`.
+   */
+  type: "RegExp";
+  /**
+   * The string representation of the regular expression value.
+   */
+  value: string;
+}
+
+/**
+ * A representation of a {@linkcode Set} Deno KV value. The value is an array
+ * of the JSON serialized values of the set.
+ */
+export interface KvSetJSON {
+  /**
+   * The type of the value, which is always `"Set"`.
+   */
+  type: "Set";
+  /**
+   * The JSON serialized version of the set values.
+   */
+  value: readonly KvValueJSON[];
+}
+
+/** Used internally to identify a typed array. */
+export type TypedArray =
```

Comment:
Can we not export this as public type?

Reply:
Previously I didn't and will amend, but the `@std` pipeline was complaining about them, so I had to export.